Extension/src/LanguageServer/ui.ts

```diff
@@ -233,27 +334,34 @@ export class UI {
         if (this.flameTimeout) {
             clearTimeout(this.flameTimeout);
         }
-        if (show && (this.isUpdatingIntelliSense || this.isRunningCodeAnalysis)) {
-            this.flameTimeout = setTimeout(() => { this.intelliSenseStatusBarItem.show(); }, this.iconDelayTime);
-        } else {
-            this.flameTimeout = setTimeout(() => { this.intelliSenseStatusBarItem.hide(); }, this.iconDelayTime);
+        if (this.intelliSenseStatusBarItem) {
+            this.intelliSenseStatusBarItem.busy = show && (this.isUpdatingIntelliSense || this.isRunningCodeAnalysis);
+            if (!this.intelliSenseStatusBarItem.busy) {
+                this.flameTimeout = setTimeout(() => {
+                    if (this.intelliSenseStatusBarItem) {
+                        this.intelliSenseStatusBarItem.dispose();
+                        this.intelliSenseStatusBarItem = undefined;
+                    }
+                }, this.iconDelayTime);
+            }
         }
     }
 
     private set ShowReferencesIcon(show: boolean) {
-        if (show && this.ReferencesCommand !== ReferencesCommandMode.None) {
-            this.referencesStatusBarItem.show();
-        } else {
-            this.referencesStatusBarItem.hide();
+        if (this.referencesStatusBarItem) {
+            this.referencesStatusBarItem.busy = show;
+            if (!show) {
+                this.referencesStatusBarItem.dispose();
+                this.referencesStatusBarItem = undefined;
+            }
         }
     }
 
     private set ShowConfiguration(show: boolean) {
-        if (show) {
-            this.configStatusBarItem.show();
-        } else {
-            this.configStatusBarItem.hide();
-        }
+        // Not needed. Goes away when not needed due to documentFilters
```

Comment:
Can ShowConfiguration be deleted?

Reply:
I think so. Seems like it's only used for disappearing the status icon when the current viewed file type changes. I'm waiting for a second opinion